src/ray/core_worker/experimental_mutable_object_provider.cc

```diff
@@ -184,19 +187,20 @@ void MutableObjectProvider::PollWriterClosure(
   RAY_CHECK(object->GetData());
   RAY_CHECK(object->GetMetadata());
 
-  reader->PushMutableObject(
-      object_id,
-      object->GetData()->Size(),
-      object->GetMetadata()->Size(),
-      object->GetData()->Data(),
-      [this, &io_context, object_id, reader](const Status &status,
-                                             const rpc::PushMutableObjectReply &reply) {
-        io_context.post(
-            [this, &io_context, object_id, reader]() {
-              PollWriterClosure(io_context, object_id, reader);
-            },
-            "experimental::MutableObjectProvider.PollWriter");
-      });
+  for (const auto &reader : readers) {
```

Comment:
@jackhumphries Is PushMutableObject synchronous? I assume it is synchronous here, so I call io_context.post after the loop so that a single IO thread can handle multiple remote reader nodes. If it is async, I may need to handle the logic in the callback function.

Reply:
I believe it's asynchronous since there is a callback function specified. If it were synchronous, we wouldn't need a callback function.

Reply:
You are correct. I traced the code, and https://github.com/ray-project/ray/blob/1dffb4d3e308324d7cdfef911b6bb5eb9f752794/src/ray/rpc/grpc_client.h#L35 shows it is an async operation.